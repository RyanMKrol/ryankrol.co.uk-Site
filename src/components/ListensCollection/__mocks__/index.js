import ListensCollection from '../ListensCollection';

const FULL_OUTPUT_MOCK = <ListensCollection outputSize="full" />;
const REDUCED_OUTPUT_MOCK = <ListensCollection outputSize="reduced" />;

const EXAMPLE_LISTENS_MOCK_ONE = {
  albumLink: 'test-albumLink-one',
  albumName: 'test-albumName-one',
  thumbnail: 'test-thumbnail-one',
  artist: 'test-artist-one',
  playcount: 1,
};

const EXAMPLE_LISTENS_MOCK_TWO = {
  albumLink: 'test-albumLink-two',
  albumName: 'test-albumName-two',
  thumbnail: 'test-thumbnail-two',
  artist: 'test-artist-two',
  playcount: 2,
};

export {
  FULL_OUTPUT_MOCK,
  REDUCED_OUTPUT_MOCK,
  EXAMPLE_LISTENS_MOCK_ONE,
  EXAMPLE_LISTENS_MOCK_TWO,
};
