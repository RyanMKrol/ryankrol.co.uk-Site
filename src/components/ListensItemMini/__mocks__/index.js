import ListensItemMini from '../ListensItemMini';

const DEFAULT_MOCK = (
  <ListensItemMini
    albumLink="https://www.last.fm/music/Kendrick+Lamar/DAMN."
    albumName="DAMN"
    thumbnail="https://lastfm.freetls.fastly.net/i/u/300x300/243a7444e2d59ff9b38e123f7267e3be.jpg"
  />
);

export default DEFAULT_MOCK;
