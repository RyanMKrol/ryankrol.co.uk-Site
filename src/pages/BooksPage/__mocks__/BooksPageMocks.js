import BooksPage from '../BooksPage';

const FULL_SIZE_MOCK = <BooksPage fullSize={true} />;
const LIMITED_SIZE_MOCK = <BooksPage fullSize={false} />;

export { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK };
