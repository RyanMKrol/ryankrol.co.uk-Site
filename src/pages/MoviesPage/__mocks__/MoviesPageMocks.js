import MoviesPage from '../MoviesPage';

const FULL_SIZE_MOCK = <MoviesPage fullSize={true} />;
const LIMITED_SIZE_MOCK = <MoviesPage fullSize={false} />;

export { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK };
