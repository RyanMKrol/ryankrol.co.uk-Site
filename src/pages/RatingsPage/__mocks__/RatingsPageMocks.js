import RatingsPage from '../RatingsPage';

const FULL_SIZE_ALBUMS_MOCK = (
  <RatingsPage
    showMovieRatings={false}
    showAlbumRatings={true}
    showTelevisionRatings={false}
    fullMovieRatings={false}
    fullAlbumRatings={true}
    fullTelevisionRatings={false}
  />
);

const LIMITED_SIZE_ALBUMS_MOCK = (
  <RatingsPage
    showMovieRatings={false}
    showAlbumRatings={true}
    showTelevisionRatings={false}
    fullMovieRatings={false}
    fullAlbumRatings={false}
    fullTelevisionRatings={false}
  />
);

const FULL_SIZE_MOVIES_MOCK = (
  <RatingsPage
    showMovieRatings={true}
    showAlbumRatings={false}
    showTelevisionRatings={false}
    fullMovieRatings={true}
    fullAlbumRatings={false}
    fullTelevisionRatings={false}
  />
);

const LIMITED_SIZE_MOVIES_MOCK = (
  <RatingsPage
    showMovieRatings={true}
    showAlbumRatings={false}
    showTelevisionRatings={false}
    fullMovieRatings={false}
    fullAlbumRatings={false}
    fullTelevisionRatings={false}
  />
);

const FULL_SIZE_TELEVISION_MOCK = (
  <RatingsPage
    showMovieRatings={false}
    showAlbumRatings={false}
    showTelevisionRatings={true}
    fullMovieRatings={false}
    fullAlbumRatings={false}
    fullTelevisionRatings={true}
  />
);

const LIMITED_SIZE_TELEVISION_MOCK = (
  <RatingsPage
    showMovieRatings={false}
    showAlbumRatings={false}
    showTelevisionRatings={true}
    fullMovieRatings={false}
    fullAlbumRatings={false}
    fullTelevisionRatings={false}
  />
);

export {
  FULL_SIZE_ALBUMS_MOCK,
  LIMITED_SIZE_ALBUMS_MOCK,
  FULL_SIZE_MOVIES_MOCK,
  LIMITED_SIZE_MOVIES_MOCK,
  FULL_SIZE_TELEVISION_MOCK,
  LIMITED_SIZE_TELEVISION_MOCK,
};
