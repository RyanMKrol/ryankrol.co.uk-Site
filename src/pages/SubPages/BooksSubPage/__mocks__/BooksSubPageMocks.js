import BooksSubPage from '../BooksSubPage';

const FULL_SIZE_MOCK = <BooksSubPage fullSize={true} />;
const LIMITED_SIZE_MOCK = <BooksSubPage fullSize={false} />;

export { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK };
