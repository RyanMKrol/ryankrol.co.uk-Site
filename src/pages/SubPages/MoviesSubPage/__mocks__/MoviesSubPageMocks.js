import MoviesSubPage from '../MoviesSubPage';

const FULL_SIZE_MOCK = <MoviesSubPage fullSize={true} />;
const LIMITED_SIZE_MOCK = <MoviesSubPage fullSize={false} />;

export { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK };
