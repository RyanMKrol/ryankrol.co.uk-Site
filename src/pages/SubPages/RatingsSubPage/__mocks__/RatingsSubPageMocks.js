import RatingsSubPage from '../RatingsSubPage';

const FULL_SIZE_ALBUMS_MOCK = (
  <RatingsSubPage
    showMovieRatings={false}
    showAlbumRatings={true}
    fullMovieRatings={false}
    fullAlbumRatings={true}
  />
);

const LIMITED_SIZE_ALBUMS_MOCK = (
  <RatingsSubPage
    showMovieRatings={false}
    showAlbumRatings={true}
    fullMovieRatings={false}
    fullAlbumRatings={false}
  />
);

const FULL_SIZE_MOVIES_MOCK = (
  <RatingsSubPage
    showMovieRatings={true}
    showAlbumRatings={false}
    fullMovieRatings={true}
    fullAlbumRatings={false}
  />
);

const LIMITED_SIZE_MOVIES_MOCK = (
  <RatingsSubPage
    showMovieRatings={true}
    showAlbumRatings={false}
    fullMovieRatings={false}
    fullAlbumRatings={false}
  />
);
export {
  FULL_SIZE_ALBUMS_MOCK,
  LIMITED_SIZE_ALBUMS_MOCK,
  FULL_SIZE_MOVIES_MOCK,
  LIMITED_SIZE_MOVIES_MOCK,
};
