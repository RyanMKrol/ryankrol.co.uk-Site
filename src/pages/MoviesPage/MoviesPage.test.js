import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK } from './__mocks__/MoviesPageMocks';

it('renders MoviesPage correctly with full size', () => {
  const header = shallow(FULL_SIZE_MOCK);
  expect(toJson(header)).toMatchSnapshot();
});
it('renders MoviesPage correctly with reduced size', () => {
  const header = shallow(LIMITED_SIZE_MOCK);
  expect(toJson(header)).toMatchSnapshot();
});
