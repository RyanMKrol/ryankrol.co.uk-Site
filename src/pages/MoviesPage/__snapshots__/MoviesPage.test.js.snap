// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MoviesPage correctly with full size 1`] = `
<div
  className="container"
>
  <h2>
    My Films
  </h2>
   
  <br />
  <p>
    Literally just a bunch of movies I own, idk
  </p>
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/movies"
    itemTag={[Function]}
  />
</div>
`;

exports[`renders MoviesPage correctly with reduced size 1`] = `
<div
  className="container"
>
  <h2>
    My Films
  </h2>
   
  <br />
  <p>
    Literally just a bunch of movies I own, idk
  </p>
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/movies"
    itemTag={[Function]}
    outputSize={20}
  />
  <Link
    className="seeMore"
    to="/MoviesList"
  >
    <h2>
      ...
    </h2>
  </Link>
</div>
`;
